Add overlay template part selector to Navigation block (behind experiment)

getdave opened the pull request:
## What

Adds the ability to select an overlay template part from the Navigation block sidebar. When an overlay template part is selected, users can navigate to edit it directly from the block editor. When an overlay is selected, clicking the overlay toggle button in the canvas navigates to the template part editor instead of toggling the menu.

## Why

This enables users to customize navigation overlays by editing dedicated overlay template parts, rather than relying solely on Navigation block controls. This aligns with the block-based approach for overlay customization, allowing styling to be managed within the template part itself.

## How

* Adds `overlayTemplatePart` attribute to Navigation block
* Creates `OverlayTemplatePartSelector` component for selecting overlay template parts
* Adds new "Overlay" panel in Navigation block sidebar
* Updates `ResponsiveWrapper` to navigate to template part editor when overlay is selected
* Integrated with experiment flag `window.__experimentalNavigationOverlays`
* Extracts reusable components (`OverlayVisibilityControl`, `OverlayMenuPreviewButton`, `OverlayMenuPreviewControls`) to avoid code duplication between the Display panel (when experiment is off) and Overlay panel (when experiment is on)

## Notes

### UI/UX Considerations

* Edit button only appears when an overlay is selected
* Edit button width set to `fit-content` for better visual appearance
* Help message: "No overlays found. Create one?"
* The "Overlay" panel remains visible even when "Overlay Visibility" is set to "Off", allowing users to re-enable it

### Future Considerations

* Preview of selected template part (out of scope for this PR)
* Ability to create a new overlay if one doesn't already exist (out of scope for this PR)

## Testing Instructions

### With Experiment Enabled

1. Enable the experiment: Set `window.__experimentalNavigationOverlays = true` in the browser console
2. Add a Navigation block to a page
3. In the block sidebar, find the "Overlay" panel
4. Use the "Overlay template" selector to choose an overlay template part
5. Click the "Edit" button to navigate to the template part editor
6. When an overlay is selected, clicking the overlay toggle button in the canvas should navigate to the template part editor instead of toggling the menu

### With Experiment Disabled

1. Ensure the experiment is disabled: `window.__experimentalNavigationOverlays` should be `undefined` or `false`
2. Add a Navigation block to a page
3. Verify that overlay controls appear in the "Display" panel (as per trunk behavior)
4. Verify all overlay functionality works as expected (overlay visibility toggle, preview controls, etc.)



Part of https://github.com/WordPress/gutenberg/issues/73080

getdave committed:
Add overlay template part selector to Navigation block

- Add overlayTemplatePart attribute to Navigation block.json
- Create OverlayTemplatePartSelector component with template part querying
- Add Edit button to navigate to focused template part editor
- Create OverlayPanel component wrapping selector in PanelBody
- Integrate panel into Navigation block inspector controls
- Gate panel behind gutenberg-customizable-navigation-overlays experiment
- Add comprehensive integration tests for selector component

The panel only appears when:
- The experiment is enabled (window.__experimentalNavigationOverlays)
- Overlay menu is not set to 'never'

getdave committed:
Simplify terminology: use 'overlay' instead of 'overlay template part'

Update all user-facing strings to use simpler 'overlay' terminology
instead of the technical 'overlay template part' phrase. This makes
the UI more user-friendly by hiding implementation details.

- Change 'Overlay Template Part' label to 'Overlay'
- Update loading message to 'Loading overlays…'
- Simplify help text and aria-labels
- Update all test strings to match new terminology

getdave committed:
Improve Edit button styling and visibility

- Only show Edit button when an overlay is selected
- Change button width from full width to fit-content
- Remove inline styles and use CSS class instead
- Add proper spacing using design system variables

getdave committed:
Fix Edit button navigation to template part editor

getdave committed:
Update overlay selector labels and help text

getdave committed:
Move overlay controls to Overlay panel and fix preview controls

- Move overlay visibility toggle and preview to Overlay panel when experiment enabled
- Fix overlay menu preview controls to work outside ToolsPanel context
- Keep panel visible when overlay is off so users can toggle it back on
- Rename 'Overlay Menu' to 'Overlay Visibility'
- Rename 'Overlay' selector to 'Overlay template'

getdave committed:
Add navigation to template part editor from overlay toggle button

getdave committed:
Update tests to use fuzzy matchers for edit button

getdave committed:
Extract duplicated overlay controls into reusable components

getdave committed:
Use setAttributes directly in overlay control components

getdave committed:
Fix linting errors: remove unused imports and fix JSDoc alignment

getdave committed:
Update docs

mtias commented on packages/block-library/src/navigation/block.json:
```diff
@@ -54,6 +54,9 @@
 			"type": "string",
 			"default": "mobile"
 		},
+		"overlayTemplatePart": {
```
I'd probably suggest just naming this attribute `overlay` or `overlayTemplate`, dropping the "Part" from the name

mtias commented on packages/block-library/src/navigation/edit/index.js:
```diff
@@ -288,6 +287,21 @@ function Navigation( {
 
 	const blockEditingMode = useBlockEditingMode();
 
+	// Get onNavigateToEntityRecord from block editor settings
```
this comment is probably just stating the obvious :)

mtias commented on packages/block-library/src/navigation/edit/index.js:
```diff
@@ -288,6 +287,21 @@ function Navigation( {
+	}, [] );
+
+	// Check if the customizable navigation overlays experiment is enabled
+	// The experiment sets window.__experimentalNavigationOverlays = true
```
likewise, just repeats the code

mtias commented on packages/block-library/src/navigation/edit/overlay-template-part-selector.js:
```diff
@@ -0,0 +1,179 @@
+
+	if ( isResolving && ! hasResolved ) {
+		return (
+			<div>
```
does this div need the same class as `wp-block-navigation__overlay-template-part-selector` applied?

getdave committed:
Resolcve review comments

mikachan commented on packages/block-library/src/navigation/edit/overlay-template-part-selector.js:
```diff
@@ -0,0 +1,179 @@
+					onClick={ handleEditClick }
+					disabled={ isEditButtonDisabled }
+					accessibleWhenDisabled
+					aria-label={
```
This `aria-label` uses the slug of the template part rather than the title, which is probably not as easy to parse as the title would be in most cases:

<img width="575" height="47" alt="image" src="https://github.com/user-attachments/assets/21bafe41-069d-4731-95b2-81e58564ecb8" />

Is it straightforward to use the title here rather than the slug?

getdave replied:
Yes. That's a mistake. Thanks for spotting!

mikachan commented on packages/block-library/src/navigation/edit/overlay-template-part-selector.js:
```diff
@@ -0,0 +1,179 @@
+				accessibleWhenDisabled
+				help={
+					overlayTemplateParts.length === 0 && hasResolved
+						? __( 'No overlays found. Create one?' )
```
I'm wondering if this line should just say "No overlays found.", as there is no link or action associated with "Create one?". Or we could improve this in a follow-up if we're going to add a link here.

getdave replied:
Yes sorry that will be in a follow up. That said, it's not obvious now so we should do what you said and enhance in the follow up.

mikachan reviewed (commented):
This is testing well for me so far 🙌 I've a couple of small suggestions on things I noticed while testing.

getdave committed:
Fix overlay template part selector: use title in aria-label and update help text

- Use template part title instead of slug in Edit button aria-label
- Change help text from 'No overlays found. Create one?' to 'No overlays found.'
- Update test to match new help text

mikachan reviewed (approved):
Thanks for addressing the feedback! This is testing well in both these scenarios:

- With Experiment Enabled ✅ 
- With Experiment Disabled ✅ 

I think we can bring this in and start iterating.
